File: app/src/main/java/com/example/android/sunshine/app/index/IndexContract.java
package com.example.android.sunshine.app.index;

import java.util.Date;
import java.util.List;

/**
 * This file is a part of the Sunshine-Version-2 project.
 */
public final class IndexContract {
    private IndexContract() {}

    public interface Display {
        void refresh();
    }

    public interface Interaction {
        void didPressRefresh();
        void didChooseForecast(int index);
    }

    public interface Synchronization {
        void bind(Display view);
        void getForecasts();
        void gotForecasts(List<Forecast> xs);
    }

    public interface Forecast {
        Date getDate();
        String getWeather();
        int getTemperature();
        int getHumidity();
    }
}

File: solshine/src/main/java/ph/codeia/solshine/index/IndexContract.kt
package ph.codeia.solshine.index

import java.util.*

/**
 * This file is a part of the Sunshine-Version-2 project.
 */
object IndexContract {
    interface Display {
        fun refresh()
    }

    interface Interaction {
        fun didPressRefresh()
        fun didChooseItem(index: Int)
        fun didPressViewMap()
    }

    interface Synchronization {
        fun bind(view: Display?)
        fun fetchForecasts()
        fun forecastsFetched(newItems: List<WeatherData>, location: Location)
    }

    interface WeatherData {
        val status: String
        val date: Date
        val minTemp: Double
        val maxTemp: Double
    }

    interface Location {
        val city: String
        val country: String
        val coords: Pair<Double, Double>
    }
}
